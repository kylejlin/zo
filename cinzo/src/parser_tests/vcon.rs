use super::*;

#[test]
fn vcon() {
    let src = r#"
(
vcon

(
    ind

    Type0

    "Nat"

    // Index types
    ()

    (
        (() ())
        ((0) ())
    )
)

// Variant constructor index - THIS MUST BE A NUMBER LITERAL.
0
)"#;
    let tokens = lex(src).unwrap();
    let cst = parse(tokens).unwrap();
    insta::assert_debug_snapshot!(&cst);
}

#[test]
fn vcon_nonliteral_ind() {
    let src = r#"
    (
    vcon
    
    0
    
    0
    )"#;
    let tokens = lex(src).unwrap();
    let err = parse(tokens).unwrap_err();
    insta::assert_debug_snapshot!(&err);
}

#[test]
fn vcon_nonliteral_vcon_index() {
    let src = r#"
(
vcon

(
    ind

    Type0

    "Nat"

    // Index types
    ()

    (
        (() ())
        ((0) ())
    )
)

(2 3)
)"#;
    let tokens = lex(src).unwrap();
    let err = parse(tokens).unwrap_err();
    insta::assert_debug_snapshot!(&err);
}
