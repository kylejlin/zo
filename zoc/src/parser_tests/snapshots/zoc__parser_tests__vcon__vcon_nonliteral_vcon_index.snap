---
source: src/parser_tests/vcon.rs
expression: "&err"
---
Some(
    LParen(
        ByteIndex(
            117,
        ),
    ),
)
